src/arch/xtensa/include/arch/compiler_info.h

```diff
@@ -29,6 +29,11 @@
 #define CC_MINOR __GNUC_MINOR__
 #define CC_MICRO __GNUC_PATCHLEVEL__
 #define CC_NAME "GCC"
+
+#if CC_MAJOR >= 10
+#define CC_USE_LIBC
+#endif
```

Comment:
gcc8 -O3 injects `memset()` calls exactly like gcc10 does and fails #3880 exactly like gcc10 does. So this needs at least a clear comment that this test is to make the difference between OUR gcc8 toolchain that is missing `-lc` versus OUR gcc10 toolchain that does have `-lc`

Same in CMakeLists.txt above.

Reply:
-O3 should not be used with xtensa GCC currently due to past subtle optmisation issues (which may now be fixed with GCC10). I dont really want O3 being used on older GCCs.

Reply:
Optimizations are not predictable. This is not about -O3 specifically, -O3 is just an example. Some source code change could trigger the same optimization and `memset()` injection with gcc8 -O2

We just need a short comment that says something "unfortunately our gcc8 toolchain missed `-lc`, please pray that it does not decides to use `memset` or other for some reason". So if that ever happens to someone's branch then they're not in the dark.

Reply:
In fact this should be:
```
if CC_MAJOR > 8.1.0
```
because any version > 8.1.0 should have the same `-lc` fix.

Reply:
To date these are not issues and we have been using GCC8 for a long time. I will be updating docs to use GCC10 for v1.8.

Reply:
We have been using _8.1.0_ for a long time. We have never used 8.2, 8.3, ... , 9.0, 9.1,...

`if CC_MAJOR > 8.1.0` makes sure anyone who tries to use any of these won't be able to use a configuration of these as broken as our first gcc10 attempt and won't waste as much time as we just did.

Reply:
> We have been using _8.1.0_ for a long time. We have never used 8.2, 8.3, ... , 9.0, 9.1,...

I and others have been using newer GCCs - I dont want to change any behaviour. 

> 
> `if CC_MAJOR > 8.1.0` makes sure anyone who tries to use any of these won't be able to use a configuration of these as broken as our first gcc10 attempt and won't waste as much time as we just did.

GCC10 is the first package that also includes C++ support for external module bindings.and newlibs is part of the GCC build now (it was separate before).  Every GCC before all used the same build config.